### PR TITLE
Revert batch 2 of the go update.

### PR DESCRIPTION
**Pull request by mattmoor**
ref: https://github.com/wolfi-dev/os/pull/6131

### CODE REVIEW

**Comment by mattmoor** [comment]
As with the prior revert, we SHOULD NOT MERGE this if we can avoid it.  If https://github.com/wolfi-dev/os/actions/runs/6438771760 passes then we should close this, but I'm staging this to prepare for a scenario where that still fails.

**Review by imjasonh — APPROVED** [review]

**Comment by dlorenc** [comment]
Do we need to merge this one? It looks like that other run Matt linked did pass?

**Pull request closed without merging**
